### DISCUSSION TITLE
自定义物料，发布到了npm,在画布能正常显示，但是预览的时候空白

### DISCUSSION THREAD

**Discussion by w990752800** [opentiny/tiny-engine#404] [Q&A]
![image](https://github.com/opentiny/tiny-engine/assets/23731304/31ece7be-86d2-44a5-9c8a-55d72e1ed509)
![image](https://github.com/opentiny/tiny-engine/assets/23731304/b52acdc3-3b7a-496b-a689-a4277e977292)

**Comment by w990752800** [comment]
物料使用的 https://unpkg.com/

**Reply by chilingling** [reply]
@w990752800 
1. 当前环境使用的是 mockServer 还是正式开源的部署后台？
2. 使用右上角调试模式按钮看看页面出码
![image](https://github.com/opentiny/tiny-engine/assets/26962197/d75254cb-9591-446c-93dd-df5bacbab491)

**Reply by w990752800** [reply]
我在本地 使用 mockService的时候没有问题； 我现在使用的 data-center, webService 后端服务启动的项目

**Reply by w990752800** [reply]
![image](https://github.com/opentiny/tiny-engine/assets/23731304/a58992e5-ee34-4458-a1a1-d8e2acda6b9d)

**Reply by chilingling** [reply]
自定义物料需要在物料配置处提前配置运行时地址
```json
{
  "componentName": "TinyButton",
  "npm": {
     "package": "@opentiny/vue",
     "script": "esm 产物地址",
     "exportName": "Button"
   }
}
```

页面出码预览应该看到出码：

```javascript
import { Button as TinyButton } form "@opentiny/vue"
```
@w990752800

**Reply by w990752800** [reply]
目前看，是所有的样式丢失了，原有的element组件，也没有样式： 
![image](https://github.com/opentiny/tiny-engine/assets/23731304/fc03db72-d93e-4be2-b17f-e416c96e8045)

**Reply by chilingling** [reply]
![image](https://github.com/opentiny/tiny-engine/assets/26962197/deaf5e05-b465-42d7-bad8-7499f0d501f0)
这里应该生成 `import { xxxx } from 'element-plus'`

**Reply by w990752800** [reply]
![image](https://github.com/opentiny/tiny-engine/assets/23731304/58910df6-9606-43f0-aedb-4d2df202c6af)
是因为我的物料没有同步到数据库？

**Reply by w990752800** [reply]
这个是需要在 tiny-engine 工程下创建 .env 然后添加sql配置？？？

**Reply by w990752800** [reply]
![image](https://github.com/opentiny/tiny-engine/assets/23731304/f64444e1-bfa6-4b2d-af6c-f214ea944e12)

**Reply by w990752800** [reply]
为啥没有生成对应的引用？？是我哪里没有配置？
